src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -469,7 +469,13 @@ impl Visitor<'tcx> for Validator<'_, 'mir, 'tcx> {
 
                     // Otherwise proceed normally: check the legality of a mutable borrow in this
                     // context.
-                    _ => self.check_op(ops::MutBorrow(kind)) == CheckOpResult::Forbidden,
+                    _ => {
+                        if let BorrowKind::Mut{ .. } = kind {
+                            self.check_op(ops::MutBorrow) == CheckOpResult::Forbidden
```

Comment:
Can you refactor out the common parts with the branch below (only the argument is different)? (also, a `match` is probably nicer here)

Reply:
Will do :)

Reply:
Hmmm well `MutBorrow` and `CellBorrow` are different types, maybe this is a good compromise:
```rust
                    // Otherwise proceed normally: check the legality of a mutable borrow in this
                    // context.
                    _ => {
                        let check_result = match kind {
                            BorrowKind::Mut{ .. } => self.check_op(ops::MutBorrow),
                            _ => self.check_op(ops::CellBorrow),
                        };
                        check_result == CheckOpResult::Forbidden
                    }
```

Reply:
ah woops; ^-- is good.

Reply:
great :) I'll push it